src/sonar_widget/SonarPlot.cc

```diff
@@ -142,3 +148,63 @@ void SonarPlot::rangeChanged(int value)
     range = value;
     drawOverlay();
 }
+
+void SonarPlot::sonarPaletteChanged(int index){
+    colorMapID = (Colormap) index;
+}
+
+void SonarPlot::generateBearingTable(int beams, base::Angle bhoriz, base::Angle bvert) {
+    bearingTable.clear();
+
+    double factor;
+    bhoriz.getDeg() == 120.0 && bvert.getDeg() == 20.0 ? factor = 0.86602540 : factor = 1.0;
+
+    for (int i = 0; i < beams; i++) {
+        double rad = asin(((2 * i - beams) * 1.0 / beams) * factor);
+        const base::Angle new_angle = base::Angle::fromRad(rad);
+        bearingTable.push_back(new_angle);
+
+    }
+}
+
+void SonarPlot::generateColormaps() {
```

Comment:
I searched already two times a function which creates heatMap like representations.
I not say that opencv the way to go is.

The simplest solution would be to move the presented code to frame-helper, would this be fine for you in regards to the overhead?